### PR TITLE
Improve build system

### PR DESCRIPTION
**Pull request by hlyytine**
Remember to test and merge with tiiuae/tii-sel4-vm#19 and tiiuae/tii_sel4_manifest#2.

### CODE REVIEW

**Review by elmankku — APPROVED** [review]
Nice, simple and efficient :+1:

File: scripts/functions.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+# container is skipped.
+if [ -z "${container}" ] && [ ! -f /.dockerenv ]; then
+  # shellcheck disable=SC2068
+  exec docker/enter_container.sh "$(pwd)" scripts/${SCRIPT_BASENAME} $@
+fi
+
+# shellcheck disable=SC1091
+. `pwd`/.config
+
+CMAKE_FLAGS=`sed -e 's/#.*$//g' < .config | xargs -iconfig -- echo -Dconfig | xargs`
```
Comment (elmankku) [inline scripts/functions.sh]: xargs man pages says `-i` is deprecated and should use `-I` instead.

```suggestion
CMAKE_FLAGS=`sed -e 's/#.*$//g' < .config | xargs -I config -- echo -Dconfig | xargs`
```

**Review by elmankku — CHANGES_REQUESTED** [review]

**Comment by hlyytine** [comment]
Found out I had pushed the version in which I had "fixed" the `make` automatic variable with the wrong one. `$<` is correct.

**Review by elmankku — APPROVED** [review]
Good

**Comment by hlyytine** [comment]
I hope I managed to write it in a self-documenting way? Also mentioning here that we shall remember to test and merge with tiiuae/tii-sel4-vm#19 and tiiuae/tii_sel4_manifest#2.

**Pull request merged**